RDBM/Table.cpp

```diff
@@ -128,33 +125,50 @@ Table& Table::AddRecord(std::initializer_list<std::string> list)
 	table.push_back(rec);
 	return *this;
 }
-void Table::DeleteRecord(unsigned int  index)
+
+Table& Table::AddColumn(const std::string& type, const std::string& name)
+{
+	colname.push_back(name);
+	if (type == "int" || type == "integer")
+		coltype.push_back('i');
+	else if (type == "double")
+		coltype.push_back('d');
+	else
+		coltype.push_back('s');
+	for (unsigned int i = 0; i < table.size(); i++)
+	{
+		if (type == "int" || type == "integer")
+			table[i]->Add("0");
+		else if (type == "double")
+			table[i]->Add("0.0");
+		else if (type == "string")
+			table[i]->Add("_");
+	}
+	return *this;
+}
+Table& Table::DeleteRecord(unsigned int  index)
 {
 	for (unsigned int i = 0; i<table.size(); i++)
 	{
 		if (table[i]->GetId()==index)
 			table.erase(table.begin());
 	}
+	return *this;
 }
 
-Record& Table::FindRecord(unsigned int index, std::ostream &out)
+Record* Table::FindRecord(unsigned int index, std::ostream &out)
 {
 	index--;
```

Comment:
the same thing with index--.

Reply:
what is wrong with it?there are two if blocks inside that checks index value
index--;
	if (index>table.size() || index < 0)
	{
		if (table.size() == 0)
		{
			Record *rec = new Record;
			return nullptr;
		}
	}
	else
		return table[index];